include/seqan3/core/debug_stream/debug_stream_type.hpp

```diff
@@ -124,7 +126,22 @@ class debug_stream_type
      */
     //!\brief Forwards to the underlying stream object.
     template <typename other_char_t, typename t>
+#if 0
     friend debug_stream_type<other_char_t> & operator<<(debug_stream_type<other_char_t> & s, t && v);
+#else
+    friend debug_stream_type<other_char_t> & operator<<(debug_stream_type<other_char_t> & s, t && v)
+    {
+        using t_ = std::remove_cvref_t<t>;
+
+        if constexpr(default_printer::is_printable<t_>) {
+            default_printer::print(s, v);
+        } else {
+            std::string const msg = std::string{"debug_stream has no print overload for type: "} + typeid(v).name();
+            throw std::runtime_error{msg};
```

Comment:
Of course, this can also be changed into a compiler error, like this:

```suggestion
            static_assert(std::is_same<t_, void>, "Type is not printable.");
```